components/esp_timer/src/esp_timer.c

```diff
@@ -43,12 +43,16 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 
 #if CONFIG_IDF_TARGET_ESP32
+#include "esp32/rom/ets_sys.h"
 #include "esp32/rtc.h"
 #elif CONFIG_IDF_TARGET_ESP32S2
+#include "esp32s2/rom/ets_sys.h"
 #include "esp32s2/rtc.h"
 #elif CONFIG_IDF_TARGET_ESP32S3
+#include "esp32s3/rom/ets_sys.h"
 #include "esp32s3/rtc.h"
 #elif CONFIG_IDF_TARGET_ESP32C3
+#include "esp32c3/rom/ets_sys.h"
```

Comment:
Is it necessary to include the same header to the other SoCs (aside esp32c3)? If so, why?

Reply:
For C3 is necessary, for the others I just placed them just to keep the consistency.

Reply:
If it is not necessary for the other SoCs, I suggest you consider not including it, otherwise, it may clutter the PR's history.

When another SoC needs this header, it can include in its own context. Does that make sense?

Reply:
Let me give my 5 cents in here: `esp32XX/rom/ets_sys.h` is not necessary at all. I think you can remove all entries in `esp_timer.c` regarding `ets_sys.h`. Can you re-check this?